homeassistant/components/integration/sensor.py

```diff
@@ -174,22 +174,25 @@ def _unit(self, source_unit: str) -> str:
     async def async_added_to_hass(self) -> None:
         """Handle entity which will be added."""
         await super().async_added_to_hass()
-        if state := await self.async_get_last_state():
+        if (state := await self.async_get_last_state()) is not None:
             try:
-                self._state = Decimal(state.state)
+                if state.state not in (STATE_UNKNOWN, STATE_UNAVAILABLE):
+                    self._state = Decimal(state.state)
+                else:
+                    self._attr_available = False
+                    self._state = None
             except (DecimalException, ValueError) as err:
                 _LOGGER.warning(
                     "%s could not restore last state %s: %s",
                     self.entity_id,
                     state.state,
                     err,
                 )
-            else:
-                self._attr_device_class = state.attributes.get(ATTR_DEVICE_CLASS)
-                if self._unit_of_measurement is None:
-                    self._unit_of_measurement = state.attributes.get(
-                        ATTR_UNIT_OF_MEASUREMENT
-                    )
+            self._attr_device_class = state.attributes.get(ATTR_DEVICE_CLASS)
+            if self._unit_of_measurement is None:
```

Comment:
Don't think this can be anything but None at this point?

Reply:
You are right, I just kept the existing code.

Fixed!